### PR TITLE
Implement KHR_materials_pbrSpecularGlossiness

### PR DESCRIPTION
**Pull request by Ralith**
Refer to [the spec](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Khronos/KHR_materials_pbrSpecularGlossiness/README.md). Not yet tested on real assets; seeking initial feedback.

### CODE REVIEW

File: gltf-json/Cargo.toml [hunk 1]
```diff
@@ -16,3 +16,4 @@ serde_json = "1.0"
 default = []
 names = []
 extras = []
+pbr-specular-glossiness = []
```
Comment (alteous) [inline gltf-json/Cargo.toml]: For familiarity I was considering using the Khronos convention for extension names, i.e. `KHR_materials_pbrSpecularGlossiness` in this case.

File: gltf-json/src/extensions/material.rs [hunk 2]
```diff
@@ -1,16 +1,126 @@
...
+use material::StrengthFactor;
+use {texture, Root, Path};
+
 /// The material appearance of a primitive.
 #[derive(Clone, Debug, Default, Deserialize, Serialize, Validate)]
-pub struct Material {}
+pub struct Material {
+    #[cfg(feature = "pbr-specular-glossiness")]
+    #[serde(default, rename = "pbrSpecularGlossiness", skip_serializing_if = "Option::is_none")]
+    pbr_specular_glossiness: Option<PbrSpecularGlossiness>,
```
Comment (alteous) [inline gltf-json/src/extensions/material.rs]: `pub pbr_specular_glossiness ...`

**Comment by Ralith** [comment]
Made the recommended changes and a few other improvements. I'm now trying to validate this on the duck sample, and getting a completely unhelpful "JSON error". Any ideas?

**Comment by Ralith** [comment]
Sorted out the issues. I think this is ready.

**Comment by alteous** [comment]
Thanks!

**Pull request merged**